### PR TITLE
Adding Gemini 

### LINKED ISSUE #1630: Requesting other Embedding Models
### 🚀 The feature

It would be much more easier and free of cost  if we use the Google's "models/text-embedding-004" embedding model instead of openai

### Motivation, pitch

Google provides this "models/text-embedding-004" model through Gemini Api which is free to use ( up-to an extent)
where as, openai  api is costly to use .

### PR DESCRIPTION
**Pull request by PranavPuranik**
## Description

This pr adds the Google Gemini embedder.

Fixes #1630 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [x] Unit Test
- [x] Test Script (please provide) - read me example. used config 

```
config = {
    "embedder": {
        "provider": "gemini",
        "config": {
            "model": "models/text-embedding-004"
        }
    },
    "vector_store": {
        "provider": "qdrant",
        "config": {
            "collection_name": "test",
            "embedding_model_dims": 768,
        }
    },
}
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: mem0/configs/prompts.py [hunk 1]
```diff
@@ -229,5 +229,5 @@ def get_update_memory_messages(retrieved_old_memory_dict, response_content):
     - If there is a deletion, the memory key-value pair should be removed from the memory.
     - If there is an update, the ID key should remain the same and only the value needs to be updated.
 
-    Do not return anything except the JSON format.
+    Only return the json format, do not add ``` around the json response. Start directly with {{ so I can load the json response with json.loads() function.
```
Comment (Dev-Khant) [inline mem0/configs/prompts.py:232]: Has this prompt change been tested throughly?

File: pyproject.toml [hunk 2]
```diff
@@ -25,6 +25,7 @@ sqlalchemy = "^2.0.31"
 langchain-community = "^0.2.12"
 neo4j = "^5.23.1"
 rank-bm25 = "^0.2.2"
+google-generativeai = "^0.8.1"
```
Comment (Dev-Khant) [inline pyproject.toml:28]: Hey, we want to make keep this as optional dependency

**Review by Dev-Khant — COMMENTED** [review]
Looks good, left 2 comments.

File: mem0/configs/prompts.py (hunk 1 above)
Comment (PranavPuranik) [reply mem0/configs/prompts.py]: This was a change from my other pr - #1884

I don't know how it came here, but yes it has been working with 2 models.

File: pyproject.toml (hunk 2 above)
Comment (PranavPuranik) [reply pyproject.toml]: I wanted to learn how to pass ci without adding this, 

```
E   ModuleNotFoundError: No module named 'google.generativeai'
```

Comment (Dev-Khant) [reply pyproject.toml]: @PranavPuranik You should add the library in `make` file under `install_all`

File: mem0/configs/prompts.py (hunk 1 above)
Comment (Dev-Khant) [reply mem0/configs/prompts.py]: Thanks

**Review by Dev-Khant — COMMENTED** [review]
Looks good, but you will also have to update `mint.json` to show new doc for gemini embeddings.

**Review by Dev-Khant — APPROVED** [review]

**Comment by Dev-Khant** [comment]
Thanks for the contribution @PranavPuranik

**Pull request merged**